Major build updates for RPMs

hecko opened the pull request:
Everything is purely connected to building RPMs using spec files and using cmake. The PR is easy to read. There is no simple way to split this into multiple PRs unfortunately so Im hoping this will pass the review.

* Update spec file for building RPMs on RHEL7, 8, 9 (Debian packages builds not updated to use cmake yet)
* Build RPMs reproducibly using Docker / Podman containers
* Small updates to cmake to support building on MacOS
* Updated GitHub actions for test-building RPMs and DEBs
* Fix paths to configuration files - this is a fix, not a change

RPM builds for EL7, 8, 9 pass as per: https://github.com/hecko/sems/actions/runs/14237787897/job/39900502510

lemenkov commented:
LGTM. I didn't test it but overall it looks good.

juha-h commented:
To me it looked like this could be spit to smaller PRs. For example, there was change in location of etc directory. Perhaps also Python changes could be done separately. As Debian user, I don't know anything about RMPs.

lemenkov commented:
> To me it looked like this could be spit to smaller PRs. For example, there was change in location of etc directory. Perhaps also Python changes could be done separately. As Debian user, I don't know anything about RPMs.

Yes, this should be much easier to merge.

hecko commented:
@juha-h @lemenkov I have updated the PR - i have removed the Python fix (created a separate PR), however I have left the paths changes as they correspond to the RPM spec file. If you think the original paths in files are correct Ill need to rewrite part of the cmake files and RPM spec file. In my eyes its more important to stay consistent on a package layer than on source layer.

juha-h commented:
Marcel Hecko writes:

> @juha-h @lemenkov I have updated the PR - i have removed the Python
> fix (created a separate PR), however I have left the paths changes as
> they correspond to the RPM spec file. If you think the original paths
> in files are correct Ill need to rewrite part of the cmake files and
> RPM spec file. In my eyes its more important to stay consistent on a
> package layer than on source layer.

OK with me.

hecko commented:
> OK with me.

@juha-h can you please approve the review in GitHub please :)

juha-h reviewed (approved):
OK, go ahead.
